sdk/tests/tutorials/properties/test_derived_properties.py

```diff
@@ -0,0 +1,145 @@
+import unittest
+import logging
+import json
+
+import lusid
+
+import lusid.models as models
+from utilities import InstrumentLoader
+from utilities import TestDataUtilities
+
+
+class DerivedPropertyTests(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # create a configured API client
+        api_client = TestDataUtilities.api_client()
+        cls.scopes_api = lusid.ScopesApi(api_client)
+        cls.property_definitions_api = lusid.PropertyDefinitionsApi(api_client)
+        cls.instruments_api = lusid.InstrumentsApi(api_client)
+        # load instruments from InstrumentLoader
+        instrument_loader = InstrumentLoader(cls.instruments_api)
+        cls.instrument_ids = instrument_loader.load_instruments()
+        # setup logging configuration
+        logging.basicConfig(level=logging.INFO)
+
+    def create_ratings_property(self, *ratings):
+
+        ratings = [*ratings]
+
+        for rating in ratings:
+            property_definition = models.CreatePropertyDefinitionRequest(
+                domain="Instrument",
+                scope="Test-Demo",
+                code=f"{rating}Rating",
+                display_name=f"{rating}Rating",
+                data_type_id=lusid.ResourceId(scope="system", code="number"),
+            )
+
+            try:
+                # create property definition
+                self.property_definitions_api.create_property_definition(
+                    create_property_definition_request=property_definition
+                )
+            except lusid.ApiException as e:
+                if json.loads(e.body)["name"] == "PropertyAlreadyExists":
+                    logging.info(f"Property {rating} already exists")
+                    pass
+
+    def upsert_ratings_property(self, figi, fitch_value=None, moodys_value=None):
+
+        properties = {
+            "Instrument/Test-Demo/FitchRating": fitch_value,
+            "Instrument/Test-Demo/MoodysRating": moodys_value,
+        }
+
+        # upsert property definition
+        for key in properties:
+            if properties[key] is not None:
```

Comment:
this seems a necessary given  you are creating the dict above

Reply:
I believe will need it with the way the upsert propreties function is currently structured. 

It seems to otherwise loop through a "None" entry and upsert to LUSID, which is then read as value=0. 

This means in an edge case where we only have a Fitch rating but no Moody's,  it will follow the rule moodys_then_fitch from the derivation formula, and since moodys turns to 0, it assigns value 0 rather than fitch.